backend/app/services/diagnosis_agent.py

```
@@ -0,0 +1,486 @@
+    import re
+    
+    try:
+        base_url = "http://ncpms.rda.go.kr/npmsAPI/service"
+        search_params = {
+            "apiKey": api_key, 
+            "serviceCode": "SVC03", 
+            "serviceType": "AA003", 
+            "insectKorName": official_pest_name,
+            "displayCount": "50"
+        }
+        search_resp = requests.get(base_url, params=search_params, timeout=10)
```

Ruff (0.15.10)

[warning] Async functions should not call blocking HTTP methods

(ASYNC210)